### PR TITLE
Bugfix/better python config

### PR DESCRIPTION
**Pull request by fgreen**
I think this is a safer way to pull in python libs though it's still a little clunky.

### CODE REVIEW

File: GeodeSupport.cmake [hunk 1]
```diff
@@ -16,7 +16,7 @@ function(add_python_module _name)
   target_link_libraries(
     ${_name}
     PUBLIC
-      ${PYTHON_LIBRARIES}
+      ${PYTHON_LIBRARY}
```
Comment (tdfischer) [inline GeodeSupport.cmake]: For consistency with cmake convention, can this be LIBRARIES instead of LIBRARY?

File: CMakeLists.txt [hunk 2]
```diff
@@ -32,18 +26,48 @@ else()
   message(FATAL_ERROR "Required dependency GMP not found. Set GMP_LIB_DIR and GMP_INCLUDE_DIR to inform cmake where to look")
 endif()
 
+# Find PYTHON_EXECTUABLE first based on $PATH
```
Comment (tdfischer) [inline CMakeLists.txt]: I think after this gets merged in we can split out all this code into a cmake/FindPython.cmake library

File: CMakeLists.txt [hunk 3]
```diff
@@ -32,18 +26,48 @@ else()
...
+    message(WARNING " Unable to find full path to python lib")
+    set(PYTHON_FOUND FALSE)
+  endif()
 else()
-  message(STATUS "Python not found. Python bindings will not be built.")
+  message(WARNING "No python interpreter found!")
+endif()
+
+if(NOT PYTHON_FOUND)
+  message(STATUS "Python libs not found. Python bindings will not be built.")
```
Comment (tdfischer) [inline CMakeLists.txt]: This seems to be being hit in jenkins. Is that a jenkins configuration bug with missing libraries?

File: GeodeSupport.cmake (hunk 1 above)
Comment (fgreen) [inline GeodeSupport.cmake]: Yes. I was unsure what the convention was supposed to be.

File: CMakeLists.txt (hunk 2 above)
Comment (fgreen) [inline CMakeLists.txt]: That sounds like a great idea.

**Comment by fgreen** [comment]
I split out a python cmake module and I think I've got it working correctly on my machine and Jenkins now.

**Comment by tdfischer** [comment]
LGTM :+1:

**Pull request merged**